unity-renderer/Assets/Scripts/MainScripts/DCL/Controllers/LoadingScreenV2/Tests/TestsEditMode/LoadingScreenControllerShould.cs

```diff
@@ -0,0 +1,164 @@
+        {
+            // Arrange
+            IAddressableResourceProvider  addressableProvider = new AddressableResourceProvider();
+            HintView hintViewPrefab = await addressableProvider.GetAddressable<HintView>(sourceHintViewAddressable, cancellationToken);
+            var loadingScreenHintsController = new LoadingScreenHintsController(GameObject.Instantiate(hintViewPrefab), hintRequestService);
+
+            // Create a TaskCompletionSource to wait for RequestHints to complete
+            var requestHintsCompletedTaskSource = new TaskCompletionSource<bool>();
+            loadingScreenHintsController.OnRequestHintsCompleted += () => requestHintsCompletedTaskSource.SetResult(true);
+
+            // Wait for RequestHints to complete
+            await requestHintsCompletedTaskSource.Task;
+
+            // Act
+            // Carousel started automatically on RequestHints
+
+            // Assert
+            // Check if the carousel was started
+            Assert.IsNotNull(loadingScreenHintsController.cancellationTokenSource);
+
+            // Act
+            loadingScreenHintsController.StopHintsCarousel();
+
+            // Assert
+            // Check if the carousel was stopped
+            Assert.IsNull(loadingScreenHintsController.cancellationTokenSource);
+        }
+
+        [Test]
+        public async Task CarouselNextAndPreviousHint()
+        {
+            // Arrange
+            IAddressableResourceProvider  addressableProvider = new AddressableResourceProvider();
+            HintView hintViewPrefab = await addressableProvider.GetAddressable<HintView>(sourceHintViewAddressable, cancellationToken);
+            var loadingScreenHintsController = new LoadingScreenHintsController(GameObject.Instantiate(hintViewPrefab), hintRequestService);
+
+            // Create a TaskCompletionSource to wait for RequestHints to complete
+            var requestHintsCompletedTaskSource = new TaskCompletionSource<bool>();
+            loadingScreenHintsController.OnRequestHintsCompleted += () => requestHintsCompletedTaskSource.SetResult(true);
+
+            // Wait for RequestHints to complete
+            await requestHintsCompletedTaskSource.Task;
+
+            // Act
+            loadingScreenHintsController.CarouselNextHint();
+            var nextHintIndex = loadingScreenHintsController.hintViewManager.currentHintIndex;
+
+            // Assert
+            // Check if the carousel moved to the next hint
+            Assert.AreEqual(1, nextHintIndex);
+
+            // Act
+            loadingScreenHintsController.CarouselPreviousHint();
+            var previousHintIndex = loadingScreenHintsController.hintViewManager.currentHintIndex;
+
+            // Assert
+            // Check if the carousel moved to the previous hint
+            Assert.AreEqual(0, previousHintIndex);
+
+            // Dispose
+            loadingScreenHintsController.cancellationTokenSource.Cancel();
+        }
```

Comment:
I would instead create HintViewManagerShould and test its functionality there, then we can just inject hintViewManager and mock it